packages/base/Tag/src/TagRectangular/TagRectangular.tsx

```diff
@@ -28,31 +20,38 @@ export const TagRectangular = forwardRef<HTMLDivElement, Props>(
       ...rest
     } = props
 
-    const classes = useStyles()
     const titleCase = useTitleCase(propsTitleCase)
 
     return (
-      <Chip
+      <div
         {...rest}
         ref={ref}
-        classes={{
-          root: cx(classes.root, classes[variant]),
-          label: classes.label,
-          icon: classes.icon,
-        }}
-        className={className}
+        className={twMerge(
+          'transition-none rounded-sm font-semibold h-4 inline-flex content-center align-middle flex-wrap',
+          variantsRootClasses[variant],
+          className
+        )}
         style={style}
-        icon={indicator ? <Indicator color={indicator} /> : undefined}
-        label={
-          <span
-            className={cx(classes.innerLabel, {
-              [classes.innerLabelDarkText]: variant === 'light-grey',
-            })}
-          >
-            {titleCase ? toTitleCase(children) : children}
-          </span>
-        }
-      />
+      >
+        {indicator && (
+          <div className='ml-1 flex content-center flex-wrap'>
+            <Indicator color={indicator} />
+          </div>
+        )}
+
+        <Typography
+          size='xxsmall'
+          weight='semibold'
+          noWrap
+          className={twMerge(
+            'text-white mx-1 w-max',
+            variant === 'light-grey' ? 'text-graphite-800' : ''
```

Comment:
```suggestion
            variant === 'light-grey' && 'text-graphite-800'
```
NIT: according twMerge documentation, we can use simple conditions